### PR TITLE
Support issues_template and merge_requests_template on gitlab_project

### LINKED ISSUE #603: Support issues_template and merge_requests_template on gitlab_project
### Terraform Version
```
$ terraform -v
Terraform v0.14.10
```

### Affected Resource(s)
Please list the resources as a list, for example:
- gitlab_project

### Terraform Configuration Files
```hcl
resource "gitlab_project" "example" {
  name        = "example"
  description = "My awesome codebase"

  visibility_level = "public"

  issues_template         = "foo"
  merge_requests_template = "bar"
}
```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
The fields are available in gitlab_project both as a data source and as a resource.

### Actual Behavior
They are yet to be implemented 😄

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
Only available in GitLab Premium and above.

### References
Dependencies:
- https://github.com/xanzy/go-gitlab/pull/1128

Docs:
- API: https://docs.gitlab.com/ee/api/projects.html#templates-for-issues-and-merge-requests
- Feature: https://docs.gitlab.com/ee/user/project/description_templates.html#set-a-default-template-for-merge-requests-and-issues

### PR DESCRIPTION
**Pull request by sirlatrom**
Fixes #603.

Awaits release including https://github.com/xanzy/go-gitlab/pull/1128, which has been merged.

### CODE REVIEW

**Comment by willianpaixao** [comment]
Hey @sirlatrom,
thanks for this PR! I'm in need of this feature too.

Wouldn't be necessary to update the documentation too? Maybe even one or examples?

File: gitlab/resource_gitlab_project_test.go [hunk 1]
```diff
@@ -1154,3 +1175,15 @@ resource "gitlab_project" "template-mutual-exclusive" {
...
 }
+
+func testAccGitlabProjectConfigIssueMergeRequestTemplates(rInt int) string {
+	return fmt.Sprintf(`
+resource "gitlab_project" "foo" {
+  name = "foo-%d"
+  path = "foo.%d"
+  description = "Terraform acceptance tests"
+  issue_template = "foo"
+  merge_request_template = "foo"
```
Comment (jtymes) [inline gitlab/resource_gitlab_project_test.go:1186]:
```suggestion
  merge_requests_template = "foo"
```

File: gitlab/resource_gitlab_project_test.go [hunk 2]
```diff
@@ -1154,3 +1175,15 @@ resource "gitlab_project" "template-mutual-exclusive" {
...
 	`, rInt, rInt)
 }
+
+func testAccGitlabProjectConfigIssueMergeRequestTemplates(rInt int) string {
+	return fmt.Sprintf(`
+resource "gitlab_project" "foo" {
+  name = "foo-%d"
+  path = "foo.%d"
+  description = "Terraform acceptance tests"
+  issue_template = "foo"
```
Comment (jtymes) [inline gitlab/resource_gitlab_project_test.go:1185]:
```suggestion
  issues_template = "foo"
```

**Review by jtymes — CHANGES_REQUESTED** [review]
Looking forward to having this available. Suggestions are to get the tests passing.

File: gitlab/resource_gitlab_project_test.go [hunk 3]
```diff
@@ -494,6 +496,25 @@ func TestAccGitlabProject_transfer(t *testing.T) {
...
+	var project gitlab.Project
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabProjectConfigIssueMergeRequestTemplates(rInt),
```
Comment (jtymes) [inline gitlab/resource_gitlab_project_test.go:509]: Does this need a `SkipFunc: isRunningInCE` since these options are only available in Premium+?

Comment (sirlatrom) [reply gitlab/resource_gitlab_project_test.go:509]: Yes, probably. However, I'm not at the same employer anymore and cannot help carry this any further.

**Review by timofurrer — CHANGES_REQUESTED** [review]
Please rebase to latest master and solve the conflicts.

Otherwise, I think up for a potential merge. What do you think @armsnyder @nagyv ?

**Comment by armsnyder** [comment]
Yep, this is a pretty straightforward change, but the test needs `SkipFunc: isRunningInCE` in order to pass, and it sounds like @sirlatrom is not able to continue the PR.

**Comment by timofurrer** [comment]
I've created a replacement PR with the open things at #796 and therefore closing this one.

**Pull request closed without merging**